src/ansys_sphinx_theme/search/fuse_search.py

```diff
@@ -67,18 +68,60 @@ def iterate_through_docs(self):
                 }
             )
 
+    def construct_title_breadcrumbs(self, section_title: str) -> str:
+        """Generate title breadcrumbs from the document name.
+
+        Parameters
+        ----------
+        section_title : str
+            Section title.
+
+        Returns
+        -------
+        str
+            Title breadcrumbs.
+
+        Notes
+        -----
+        This method is used to generate the title breadcrumbs for the search index.
+        The title breadcrumbs are generated by traversing the document name and
+        getting the title of the document parts.
+        """
+        # removing the last part of the doc_name because it is the file name taken from the nodes
+        docs_parts = self._doc_name.split("/")[:-1]
+
+        title_string = [
+            self.env.titles[doc_part].astext()
+            for doc_part in docs_parts
+            if doc_part in self.env.titles
+        ]
+
+        self.breadcrumbs_title = " > ".join(title_string)
+
+        if self.breadcrumbs_title:
+            if self.section_title == self.doc_title:
```

Comment:
I see what you are doing here. This is nice! However, sometimes, conditions obfuscate what we are checking for. You can isolate the condition, which allows for naming and reusing it later:

```
has_duplicated_items = self.section_title == self.doc_title
```

Reply:
Also, the logic below can be compressed to:

```suggestion
    # Join the titles with " > "
    breadcrumbs = " > ".join(title_parts)
    
    # Check whether to avoid including duplicated items
    ignore_doc_title = self.section_title == self.doc_title

    # Determine the final breadcrumb output
    if ignore_doc_title:
        breadcrumb = f"{breadcrumbs} > {section_title}" if breadcrumbs else section_title
    else:
        breadcrumb = f"{breadcrumbs} > {self.doc_title} > {section_title}" if breadcrumbs else f"{self.doc_title} > {section_title}"
    
    return breadcrumb
```